### ISSUE TITLE
`WITH` clauses are not recognized properly

### ISSUE THREAD

**Issue by pgab** [GiGainfosystems/diesel-oci#8]
https://github.com/GiGainfosystems/diesel-oci/blob/feature/more_gst_compat/src/oracle/connection/stmt.rs#L88 this can go wrong, since where is also `WITH` and other SQL structures. before
there was `sql.contains("SELECT")||sql.contains("select")` which might fails on the
following queries (meaning they will be identified as select clause even if they are
not: `UPDATE table SET k='select';` OR
```
CREATE OR REPLACE FORCE VIEW full_bounding_boxes(id, o_c1, o_c2, o_c3, u_c1, u_c2, u_c3, v_c1, v_c2, v_c3, w_c1, w_c2, w_c3)
AS
SELECT bbox.id as id,
o.c1 as o_c1, o.c2 as o_c2, o.c3 as o_c3,
u.c1 as u_c1, u.c2 as u_c2, u.c3 as u_c3,
v.c1 as v_c1, v.c2 as v_c2, v.c3 as v_c3,
w.c1 as w_c1, w.c2 as w_c2, w.c3 as w_c3
FROM bounding_boxes bbox
INNER JOIN geo_points o ON bbox.o = o.id
INNER JOIN geo_points u ON bbox.u = u.id
INNER JOIN geo_points v ON bbox.v = v.id
INNER JOIN geo_points w ON bbox.w = w.id
```

**Comment by geomaniac** [comment]
We could mitigate some of those with statement attributes. E.g. checking if a statement is `SELECT` or `INSERT`. This however won's fix the `LIMIT` and `RETURNING` issues.

E.G. `OCI_ATTR_STMT_TYPE` or `OCI_ATTR_SQLFNCODE`.

**Comment by geomaniac** [comment]
Fixed with #26
